### PR TITLE
Remove solaris build tag and `contrib/mkimage/solaris`

### PR DESCRIPTION
**Pull request by yongtang**
As Solaris is not really supported any more. I am proposing to remove the build tags and other reference to Solaris from the code base. 

This fix cleaned all traces of solaris in build tags and files.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

**Review by tianon — APPROVED** [review]
LGTM

File: libcontainerd/remote_daemon_options_unix.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build linux solaris
+// +build linux
```
Comment (thaJeztah) [inline libcontainerd/remote_daemon_options_unix.go]: this file should probably be renamed to `_linux.go` then the build-tag can be removed

File: libcontainerd/remote_daemon_process_unix.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// +build linux solaris
+// +build linux
```
Comment (thaJeztah) [inline libcontainerd/remote_daemon_process_unix.go]: same here

File: libcontainerd/remote_daemon_unix.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// +build linux solaris
+// +build linux
```
Comment (thaJeztah) [inline libcontainerd/remote_daemon_unix.go]: same here

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: libcontainerd/remote_daemon_options_unix.go (hunk 1 above)
Comment (yongtang) [reply libcontainerd/remote_daemon_options_unix.go]: Thanks that works! Updated.

File: libcontainerd/remote_daemon_process_unix.go (hunk 2 above)
Comment (yongtang) [reply libcontainerd/remote_daemon_process_unix.go]: Done.

File: libcontainerd/remote_daemon_unix.go (hunk 3 above)
Comment (yongtang) [reply libcontainerd/remote_daemon_unix.go]: Done.

**Review by thaJeztah — APPROVED** [review]
LGTM (if green :smile:)

**Pull request merged**